### PR TITLE
chore: throttle exceptions counter increment

### PR DESCRIPTION
**Pull request by sampaiodiego**
## Proposed changes (including videos or screenshots)

Throttle down uncaught exceptions counter to every 10 seconds. This is to avoid too many operations into the settings collection.

### CODE REVIEW

**Review by rodrigok — APPROVED** [review]

File: packages/tools/src/counter.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+		},
+		wait,
+		{ leading: false },
+	);
+
+	return () => {
+		counter++;
+		throttledFn();
+	};
+}
```
Comment (ggazzo) [inline packages/tools/src/counter.ts:20]: I dont think this should be placed here, its not quite useful outside that scope

not sure about the `lodash` necessity as well

File: apps/meteor/app/error-handler/server/lib/RocketChat.ErrorHandler.ts [hunk 2]
```diff
@@ -1,9 +1,14 @@
 import { Meteor } from 'meteor/meteor';
 import { Settings, Users, Rooms } from '@rocket.chat/models';
+import { throttledCounter } from '@rocket.chat/tools';
 
 import { settings } from '../../../settings/server';
 import { sendMessage } from '../../../lib/server';
 
+const incException = throttledCounter((counter) => {
```
Comment (KevLehman) [inline apps/meteor/app/error-handler/server/lib/RocketChat.ErrorHandler.ts:8]: wdyt of ignoring this setting on the `watch.settings` watcher?

File: packages/tools/src/counter.ts (hunk 1 above)
Comment (ggazzo) [inline packages/tools/src/counter.ts:20]:
```suggestion
export const throttle = <R, A extends any[]>(
    fn: (...args: A) => R,
    delay: number
) => {
    let wait = false;
    let timeout: undefined | number;

    return Object.assign(
        (...args: A) => {
            if (wait) return undefined;
  
            const val = fn(...args);
  
            wait = true;
  
            timeout = window.setTimeout(() => {
                wait = false;
            }, delay);
  
            return val;
        }, {
        stop: () => {
            clearTimeout(timeout);
        },
    });
};
```

File: apps/meteor/app/error-handler/server/lib/RocketChat.ErrorHandler.ts [hunk 3]
```diff
@@ -1,9 +1,14 @@
 import { Meteor } from 'meteor/meteor';
 import { Settings, Users, Rooms } from '@rocket.chat/models';
+import { throttledCounter } from '@rocket.chat/tools';
```
Comment (ggazzo) [inline apps/meteor/app/error-handler/server/lib/RocketChat.ErrorHandler.ts:3]:
```suggestion
import { throttle } from '@rocket.chat/tools';
```

File: apps/meteor/app/error-handler/server/lib/RocketChat.ErrorHandler.ts [hunk 4]
```diff
@@ -1,9 +1,14 @@
...
 
 import { settings } from '../../../settings/server';
 import { sendMessage } from '../../../lib/server';
 
+const incException = throttledCounter((counter) => {
+	Settings.incrementValueById('Uncaught_Exceptions_Count', counter).catch(console.error);
+}, 10000);
+
```
Comment (ggazzo) [inline apps/meteor/app/error-handler/server/lib/RocketChat.ErrorHandler.ts:11]:
```suggestion
const incException = (() => {
    let counter = 0;
    const commit = throttle((increment: number) => {
        console.log(counter);
        counter = 0;
    }, 10000);
    return () => {
        counter++;
        commit(counter);
    };
})()
```

File: apps/meteor/app/error-handler/server/lib/RocketChat.ErrorHandler.ts (hunk 2 above)
Comment (sampaiodiego) [reply apps/meteor/app/error-handler/server/lib/RocketChat.ErrorHandler.ts:8]: the main issue is that we'd still receive it from mongo.. so we'd need to remove it on oplog/change stream query, which is not good 😬

**Review by ggazzo — APPROVED** [review]

**Pull request merged**